multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -2225,6 +2235,12 @@ def predict(
                     detection_classes=self._model.model.CLASSES,
                     result_path=None,
                 )
+            elif self._problem_type == OPEN_VOCABULARY_OBJECT_DETECTION:
+                pred = save_ovd_result_df(
```

Comment:
Why do we need to save results for ovd? `result_path=None` means not saving, but the function name is still misleading. By default, we need to return the same format as input.

Reply:
We are returning dict if as_pandas is False. Here we reuse save_ovd_result_df that both formatting result to df and save the df (if result_path is not None). Later we will add save result feature for ovd (using this function) together with evaluation for ovd.

Reply:
`save_ovd_result_df ` seems to always return a panda dataframe, even though the input data is a dict.

Reply:
Yes, that's also true for object detection. We only save panda dataframe. But will return list of dicts if input is not df.